### PR TITLE
Added Monitoring Capabilities (Observability-as-code)

### PR DESCRIPTION
**Pull request by sudivate**
## Summary of the Pull Request
Added below capabilities that can be optionally disabled.
- Azure Dashboard, alerts and diagnostic settings for DataFactory and Synapse
- Log analytics Workspace and Azure Data Factory Analytics Solution
- Update Portal Experience  
- Update Doc 

## References
No reference

## PR Checklist
* [x ] Code of Conduct signed. If not, go to [Code of Conduct](https://github.com/Azure/data-management-zone/blob/main/CODE_OF_CONDUCT.md).

## Validation Steps Performed
Deployed manually

### CODE REVIEW

**Comment by sudivate** [comment]
@marvinbuss Looks like this fix is not yet released in version 0.4.1272  https://github.com/Azure/bicep/pull/6009

File: infra/params.dev.json [hunk 1]
```diff
@@ -71,6 +71,12 @@
         "enableSearch": {
             "value": true
         },
+        "enableMonitoring": {
```
Comment (bjcmit) [inline infra/params.dev.json:74]: Why is enableMonitoring set in dev, but in prod and test enableObservability is set?

File: infra/modules/services/loganalytics.bicep [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+param location string
+param tags object
+param processingService string
+
+// Variables
+var dataFactoryAnalyticsName = 'AzureDataFactoryAnalytics(${loganalyticsName})'
+
+// Resources
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2020-10-01' = {
+  name: logAnanalyticsName
```
Comment (bjcmit) [inline infra/modules/services/loganalytics.bicep:17]: This looks like a typo

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please see review below.

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please see review below.

File: infra/main.bicep [hunk 3]
```diff
@@ -138,6 +144,11 @@ var applicationInsights001Name = '${name}-insights001'
 var containerRegistry001Name = '${name}-containerregistry001'
 var storage001Name = '${name}-storage001'
 var machineLearning001Name = '${name}-machinelearning001'
+var logAnalytics001Name = '${name}-loganalytics'
```
Comment (marvinbuss) [inline infra/main.bicep:147]:
```suggestion
var logAnalytics001Name = '${name}-loganalytics001'
```

File: infra/modules/services/alerts.bicep [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+// Resources
+resource actiongroup 'Microsoft.Insights/actionGroups@2021-09-01' =  {
+  name: dataFactoryEmailActionGroup
+  location: 'global'
+  tags: tags
+  properties: {
+    groupShortName: 'emailgroup'
+    emailReceivers: [
+      {
+        emailAddress: dataProductTeamEmail
```
Comment (marvinbuss) [inline infra/modules/services/alerts.bicep:29]: What if the email address is empty? Would this still work?

**Review by marvinbuss — COMMENTED** [review]
Please see comments below. ALso, tehre seems to be some issue with ARM. The run below is failing. Can you please take a look?

**Comment by marvinbuss** [comment]
[Preview Link](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-product-analytics%2Fsudivate%2Fdata-observability%2Finfra%2Fmain.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-product-analytics%2Fsudivate%2Fdata-observability%2Fdocs%2Freference%2Fportal.dataProduct.json)

**Comment by marvinbuss** [comment]
@sudivate Something needs to be corrected here:
image

There is nothing selected by default.

File: docs/reference/portal.dataProduct.json [hunk 5]
```diff
@@ -1427,9 +1496,11 @@
...
                 "databricksWorkspaceUrl": "[if(empty(steps('generalSettings').machineLearningDeploymentSettings.databricksWorkspaceUrl.properties.workspaceUrl), '', steps('generalSettings').machineLearningDeploymentSettings.databricksWorkspaceUrl.properties.workspaceUrl)]",
                 "databricksAccessToken": "[if(empty(steps('generalSettings').machineLearningDeploymentSettings.databricksAccessToken), '', steps('generalSettings').machineLearningDeploymentSettings.databricksAccessToken)]",
-                "enableRoleAssignments": "[steps('generalSettings').generalSettings.enableRoleAssignments]",
+                "enableRoleAssignments": "[equals(steps('generalSettings').generalSettings.enableRoleAssignments, 'yes')]",
                 "cognitiveServiceKinds": "[if(empty(steps('generalSettings').generalSettings.cognitiveServiceKinds), parse('[]'), steps('generalSettings').generalSettings.cognitiveServiceKinds)]",
                 "enableSearch": "[steps('generalSettings').generalSettings.enableSearch]",
```
Comment (marvinbuss) [inline docs/reference/portal.dataProduct.json:1501]:
```suggestion
                "enableSearch": "[equals(steps('generalSettings').generalSettings.enableSearch, 'yes')]",
```

Comment (marvinbuss) [reply docs/reference/portal.dataProduct.json:1501]: This change was merged into this branch, but you removed it again. See here: https://github.com/Azure/data-product-analytics/pull/121/files

File: docs/reference/portal.dataProduct.json [hunk 6]
```diff
@@ -980,13 +988,74 @@
...
+                                        }
+                                    }
+                                },
+                                {
+                                    "name": "enableMonitoring",
+                                    "label": "Deploy Monitoring Components",
+                                    "type": "Microsoft.Common.OptionsGroup",
+                                    "visible": true,
+                                    "toolTip": "Deploy monitoring components like Log Analytics, Azure dashboards, alerts and diagnostic settings.",
+                                    "defaultValue": "yes",
```
Comment (marvinbuss) [inline docs/reference/portal.dataProduct.json:1034]:
```suggestion
                                    "defaultValue": "Yes (recommended)",
```

File: docs/reference/portal.dataProduct.json [hunk 7]
```diff
@@ -980,13 +988,74 @@
...
+                                    "label": "Alert Receivers Email ID",
+                                    "type": "Microsoft.Common.TextBox",
+                                    "visible": "[equals(steps('generalSettings').monitoringSettings.enableMonitoring, 'yes')]",
+                                    "defaultValue": "",
+                                    "toolTip": "Specify Email ID to receive monitoring alerts.",
+                                    "constraints": {
+                                        "required": true,
+                                        "regex": "^[\\S+@\\S+.\\S+]{7,100}$",
+                                        "validationMessage": "Please enter a valid email address."
                                     }
```
Comment (marvinbuss) [inline docs/reference/portal.dataProduct.json:1059]:
```suggestion
                                    "constraints": {
                                        "required": true,
                                        "validations": [
                                            {
                                                "regex": "^[\\S+@\\S+.\\S+]{7,100}$",
                                                "message": "Please enter a valid email address."
                                            }
                                        ]
                                    }
```

Comment (marvinbuss) [reply docs/reference/portal.dataProduct.json:1059]: This is the correct Syntax here.

**Review by marvinbuss — CHANGES_REQUESTED** [review]
See required changes below:

File: infra/main.bicep [hunk 8]
```diff
@@ -138,6 +144,11 @@ var applicationInsights001Name = '${name}-insights001'
 var containerRegistry001Name = '${name}-containerregistry001'
 var storage001Name = '${name}-storage001'
 var machineLearning001Name = '${name}-machinelearning001'
+var logAnalytics001Name = '${name}-loganalytics001'
+var dataFactoryEmailActionGroup = '${datafactory001Name}-${name}-emailactiongroup'
+var adfPipelineFailedAlertName = '${datafactory001Name}-${name}-adffailedalert'
```
Comment (marvinbuss) [inline infra/main.bicep:149]:
```suggestion
var adfPipelineFailedAlertName = '${datafactory001Name}-${name}-failedalert'
```

**Review by marvinbuss — COMMENTED** [review]
Just small inconsistency.

File: docs/reference/portal.dataProduct.json (hunk 5 above)
Comment (sudivate) [reply docs/reference/portal.dataProduct.json]: Included that change now

File: docs/reference/portal.dataProduct.json (hunk 7 above)
Comment (sudivate) [reply docs/reference/portal.dataProduct.json:1059]: yup

**Comment by marvinbuss** [comment]
@sudivate Can you please validate the deployment?
image
"ResourceNotFound\",\r\n    \"message\": \"The Resource 'Microsoft.DataFactory/factories/aaaaa-dev-datafactory001' under resource group 'test-rg' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\"

**Review by marvinbuss — APPROVED** [review]
LGTM!

**Pull request merged**